<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1419</width>
    <height>814</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>361</width>
      <height>171</height>
     </rect>
    </property>
    <property name="title">
     <string>Received Data</string>
    </property>
    <widget class="QTextBrowser" name="textBrowser">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>341</width>
       <height>131</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>1130</x>
      <y>490</y>
      <width>211</width>
      <height>231</height>
     </rect>
    </property>
    <property name="title">
     <string>Control Parameters</string>
    </property>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>191</width>
       <height>111</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Position</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Velocity</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Accelometer</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QLineEdit" name="posLineEdit"/>
        </item>
        <item>
         <widget class="QLineEdit" name="velLineEdit"/>
        </item>
        <item>
         <widget class="QLineEdit" name="accLineEdit"/>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="layoutWidget_4">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>140</y>
       <width>191</width>
       <height>71</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QPushButton" name="motionButton">
          <property name="text">
           <string>MOTION</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="runButton">
          <property name="text">
           <string>RUN</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QPushButton" name="senparamsButton">
          <property name="text">
           <string>SEND PARAMS</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="getButton">
          <property name="text">
           <string>GET</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_7">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>190</y>
      <width>331</width>
      <height>151</height>
     </rect>
    </property>
    <property name="title">
     <string>PID Tunning</string>
    </property>
    <widget class="QWidget" name="layoutWidget_2">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>211</width>
       <height>121</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Kp</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Ki</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>Kd</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QLineEdit" name="kpEdit"/>
        </item>
        <item>
         <widget class="QLineEdit" name="kiEdit"/>
        </item>
        <item>
         <widget class="QLineEdit" name="kdEdit"/>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="layoutWidget_3">
     <property name="geometry">
      <rect>
       <x>240</x>
       <y>20</y>
       <width>82</width>
       <height>121</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_7">
      <item>
       <widget class="QPushButton" name="sendpidButton">
        <property name="text">
         <string>SEND</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="tunningButton">
        <property name="text">
         <string>TUNING</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="requestButton">
        <property name="text">
         <string>REQUEST</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>350</y>
      <width>611</width>
      <height>411</height>
     </rect>
    </property>
    <property name="title">
     <string>PID Tunning Graph</string>
    </property>
    <widget class="QCustomPlot" name="pidPlot" native="true">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>591</width>
       <height>371</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_5">
    <property name="geometry">
     <rect>
      <x>1130</x>
      <y>10</y>
      <width>211</width>
      <height>241</height>
     </rect>
    </property>
    <property name="title">
     <string>Serial Port</string>
    </property>
    <widget class="QWidget" name="layoutWidget_5">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>191</width>
       <height>211</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_8">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>COM</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="serialComboBox"/>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>BaudRate</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="baudComboBox"/>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QLabel" name="label_9">
          <property name="text">
           <string>DataSize</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="sizeComboBox"/>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <item>
         <widget class="QLabel" name="label_10">
          <property name="text">
           <string>Parity</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="parityComboBox"/>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_6">
    <property name="geometry">
     <rect>
      <x>1130</x>
      <y>310</y>
      <width>211</width>
      <height>171</height>
     </rect>
    </property>
    <property name="title">
     <string>Send Data</string>
    </property>
    <widget class="QLineEdit" name="sendEdit">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>191</width>
       <height>51</height>
      </rect>
     </property>
    </widget>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>191</width>
       <height>71</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_13">
      <item>
       <widget class="QPushButton" name="sendButton">
        <property name="text">
         <string>Send</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="clearButton">
        <property name="text">
         <string>Clear</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>630</x>
      <y>10</y>
      <width>491</width>
      <height>751</height>
     </rect>
    </property>
    <property name="title">
     <string>Control Graph</string>
    </property>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>471</width>
       <height>711</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_10">
      <item>
       <widget class="QCustomPlot" name="posPlot" native="true"/>
      </item>
      <item>
       <widget class="QCustomPlot" name="velPlot" native="true"/>
      </item>
      <item>
       <widget class="QCustomPlot" name="accPlot" native="true"/>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_8">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>220</y>
      <width>211</width>
      <height>111</height>
     </rect>
    </property>
    <property name="title">
     <string>Tools</string>
    </property>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>191</width>
       <height>91</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_8">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_11">
        <item>
         <widget class="QPushButton" name="savePIDButton">
          <property name="text">
           <string>Print PID</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="clearPIDButton">
          <property name="text">
           <string>Clear PID</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_12">
        <item>
         <widget class="QPushButton" name="infoPIDButton">
          <property name="text">
           <string>PID Info</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="clearReceiveButton">
          <property name="text">
           <string>Clear Box</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>1130</x>
      <y>250</y>
      <width>211</width>
      <height>58</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_9">
     <item>
      <widget class="QPushButton" name="openButton">
       <property name="text">
        <string>OPEN</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="closeButton">
       <property name="text">
        <string>CLOSE</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="groupBox_9">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>10</y>
      <width>241</width>
      <height>171</height>
     </rect>
    </property>
    <property name="title">
     <string>USB</string>
    </property>
    <widget class="QWidget" name="">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>30</y>
       <width>141</width>
       <height>131</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_15">
      <item>
       <widget class="QLabel" name="portlabel">
        <property name="font">
         <font>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Not Connected!</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="vendorIDLabel">
        <property name="font">
         <font>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Not Connected!</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="productIDLabel">
        <property name="font">
         <font>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Not Connected!</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="manuLabel">
        <property name="font">
         <font>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Not Connected!</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="vendorLabel">
        <property name="font">
         <font>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Not Connected!</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>77</width>
       <height>131</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_14">
      <item>
       <widget class="QLabel" name="label_16">
        <property name="text">
         <string>Port:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_11">
        <property name="text">
         <string>VendorID:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_12">
        <property name="text">
         <string>ProductID:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_17">
        <property name="text">
         <string>Manufacturer:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_13">
        <property name="text">
         <string>Vendor:</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1419</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
